awx/ui/client/src/templates/workflows/edit-workflow/workflow-edit.controller.js

```diff
@@ -63,6 +63,179 @@ export default [
+            isModalOpen: false,
+            isModalReady: false,
+            modalSelectedId: null,
+            modalSelectedName: null,
+            modalBaseParams: {
+                order_by: 'name',
+                page_size: 5,
+                credential_type__namespace: `${workflowJobTemplateData.webhook_service}_token`,
+            },
+            modalTitle: i18n._('Select Webhook Credential'),
+        };
+
+        $scope.handleWebhookCredentialLookupClick = () => {
+            $scope.webhookCredential.modalSelectedId = $scope.webhookCredential.id;
+            $scope.webhookCredential.isModalOpen = true;
+        };
+
+        $scope.handleWebhookCredentialTagDelete = () => {
+            $scope.webhookCredential.id = null;
+            $scope.webhookCredential.name = null;
+        };
+
+        $scope.handleWebhookCredentialModalClose = () => {
+            $scope.webhookCredential.isModalOpen = false;
+            $scope.webhookCredential.isModalReady = false;
+        };
+
+        $scope.handleWebhookCredentialModalReady = () => {
+            $scope.webhookCredential.isModalReady = true;
+        };
+
+        $scope.handleWebhookCredentialModalItemSelect = (item) => {
+            $scope.webhookCredential.modalSelectedId = item.id;
+            $scope.webhookCredential.modalSelectedName = item.name;
+        };
+
+        $scope.handleWebhookCredentialModalCancel = () => {
+            $scope.webhookCredential.isModalOpen = false;
+            $scope.webhookCredential.isModalReady = false;
+            $scope.webhookCredential.modalSelectedId = null;
+            $scope.webhookCredential.modalSelectedName = null;
+        };
+
+        $scope.handleWebhookCredentialSelect = () => {
+            $scope.webhookCredential.isModalOpen = false;
+            $scope.webhookCredential.isModalReady = false;
+            $scope.webhookCredential.id = $scope.webhookCredential.modalSelectedId;
+            $scope.webhookCredential.name = $scope.webhookCredential.modalSelectedName;
+            $scope.webhookCredential.modalSelectedId = null;
+            $scope.webhookCredential.modalSelectedName = null;
+        };
+
+        $scope.handleWebhookKeyButtonClick = () => {
+            Rest.setUrl(workflowJobTemplateData.related.webhook_key);
+            Wait('start');
+            Rest.post({})
+                .then(({ data }) => {
+                    $scope.currentlySavedWebhookKey = data.webhook_key;
+                    $scope.webhook_key = data.webhook_key;
+                })
+                .catch(({ data }) => {
+                    const errorMsg = `Failed to generate new webhook key. POST returned status: ${status}`;
+                    ProcessErrors($scope, data, status, form, { hdr: 'Error!', msg: errorMsg });
+                })
+                .finally(() => {
+                    Wait('stop');
+                });
+        };
+
+        $('#content-container').append($compile(`
```

Comment:
this feels a bit strange.  could we go ahead and have this compiled (as part of the view's partial, and then just conditionally show/hide it when some portion of the data related to $scope.webhookCredential is updated?

Reply:
ah totally understand, basically I thought that there was a template to put it in, but it looks like there's not.  No problem!